src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -916,14 +914,17 @@ namespace winrt::TerminalApp::implementation
             // editors, who will write a temp file, then rename it to be the
             // actual file you wrote. So listen for that too.
             wil::FolderChangeEvents::FileName | wil::FolderChangeEvents::LastWriteTime,
-            [this, settingsBasename = settingsPath.filename(), stateBasename = statePath.filename()](wil::FolderChangeEvent, PCWSTR fileModified) {
+            [this, settingsPath](wil::FolderChangeEvent, PCWSTR fileModified) {
+                static const std::filesystem::path statePath{ std::wstring_view{ ApplicationState::SharedInstance().FilePath() } };
+                static const std::filesystem::path elevatedStatePath{ std::wstring_view{ ElevatedState::SharedInstance().FilePath() } };

```

Comment:
I personally prefer the previous code, which didn't call `.filename()` repeatedly in the closure. The `.filename()` function could possibly be a very heavy operation after all (for instance by checking the file system in future STL versions)...
However we can shorten the existing code by adding some helper lambdas:
```cpp
static constexpr auto intoPath = [](const std::wstring_view& path) -> std::filesystem::path {
    return path;
};
static constexpr auto intoFilename = [](const std::wstring_view& path) -> std::filesystem::path {
    return std::filesystem::path{ path }.filename();
};
```

Afterwards we can shorten the capture group down to a single character:
```cpp
void AppLogic::_RegisterSettingsChange()
{
    // ... lambdas here ...
    
    const auto settingsPath = intoPath(CascadiaSettings::SettingsPath());
    const auto settingsBasename = settingsPath.filename();
    const auto stateBasename = intoFilename(ApplicationState::SharedInstance().FilePath());
    const auto elevatedStateBasename = intoFilename(ElevatedState::SharedInstance().FilePath());

    _reader.create(
        // ...
        [=](wil::FolderChangeEvent, PCWSTR fileModified) {
            // ...
        });
}
```

Reply:
See, I tried just capturing them all, but the compiler gave me some error about having too much stuff being captured or something. I forget the error exactly. It was unhappy with me capturing the fourth thing. Hence moving the stuff inside the lambda itself.

Reply:
`wistd::function` only allows holding 12 pointers and a `std::filesystem::path` is 4 pointers large.
I wonder what's so bad about `std::function` that it wasn't used here. I guess the reason is that they wanted something that's `noexcept` constructible... (but why?) Certainly very annoying though.

In that case however you could still write this (outside of the closure - it's automatically captured):
```cpp
static const auto settingsBasename = settingsPath.filename();
static const auto stateBasename = intoFilename(ApplicationState::SharedInstance().FilePath());
static const auto elevatedStateBasename = intoFilename(ElevatedState::SharedInstance().FilePath());
```

Reply:
BTW I wonder if we're doing it all wrong either way (if we're thinking outside of the scope of this PR)...
Shouldn't the the caller - in this case the `AppLogic` class - decide which file to load and not `CascadiaSettings`? I guess `LoadAll` should have a path string argument. That way we could safely hardcode the correct filenames into `AppLogic.cpp`.